frontend/dynamic-demo-plugin/src/utils/telemetry.ts

```diff
@@ -0,0 +1,89 @@
+const SEGMENT_KEY = '';
+
+const initSegment = () => {
+  const analytics = ((window as any).analytics = (window as any).analytics || []);
+  if (!analytics.initialize) {
+    if (analytics.invoked)
+      window.console && console.error && console.error('Segment snippet included twice.');
+    else {
+      analytics.invoked = !0;
+      analytics.methods = [
+        'trackSubmit',
+        'trackClick',
+        'trackLink',
+        'trackForm',
+        'pageview',
+        'identify',
+        'reset',
+        'group',
+        'track',
+        'ready',
+        'alias',
+        'debug',
+        'page',
+        'once',
+        'off',
+        'on',
+        'addSourceMiddleware',
+        'addIntegrationMiddleware',
+        'setAnonymousId',
+        'addDestinationMiddleware',
+      ];
+      analytics.factory = function(e: string) {
+        return function() {
+          var t = Array.prototype.slice.call(arguments);
+          t.unshift(e);
+          analytics.push(t);
+          return analytics;
+        };
+      };
+      for (var e = 0; e < analytics.methods.length; e++) {
+        var key = analytics.methods[e];
+        analytics[key] = analytics.factory(key);
+      }
+      analytics.load = function(key: string, e: Event) {
+        const t = document.createElement('script');
+        t.type = 'text/javascript';
+        t.async = !0;
+        t.src = 'https://cdn.segment.com/analytics.js/v1/' + key + '/analytics.min.js';
```

Comment:
This is probably not an issue in practice since I wouldn't expect key to have special characters, but I'd use `encodeURIComponent` to be safe.

```suggestion
        t.src = 'https://cdn.segment.com/analytics.js/v1/' + encodeURIComponent(key) + '/analytics.min.js';
```